File: Stack implementation using an array.cpp
#include <bits/stdc++.h>
using namespace std;
#define maximum_size 10
int arr[maximum_size];
class use_for_stack
{
	int position_of_top=-1;
    public: int push(int m);
            int pop();
            int peak();
            int display();
};
use_for_stack::push(int m)
{
	if (position_of_top >= (maximum_size - 1))
    {
		cout <<"\nOverflow(insertion not possible): "<<m<<"\n";
	}
	else
	{
		arr[++position_of_top] = m;
		cout<<"\nInserted element in stack: "<< m;
	}
}

use_for_stack::pop()
{

	if (position_of_top < 0)
    {
		cout <<"Underflow(deletion not possible)\n";
	}
	else
    {
        int m = arr[position_of_top--];
        cout <<"\nPopped element from stack: "<<m;
	}
}
use_for_stack::peak()
{
    cout<<"\nTop Element in the stack : ";
    int m=0;
	if (position_of_top < 0)
    {
		cout<<"Stack is Empty";
	}
	else
	{
                m = arr[position_of_top];
                cout<<m<<" ";
	}
}
use_for_stack::display()
{

    int m=0;
	if (position_of_top < 0)
    {
		cout<<"Stack is Empty";
	}
	else
	{       cout<<"\nElements in the stack : ";
            while(position_of_top>=0)
            {
                m = arr[position_of_top--];
                cout<<m<<" ";
            }
	}
}
int main()
{
	class use_for_stack stack;
    stack.push(10);
    stack.push(11);
	stack.push(12);
	stack.push(13);
    stack.push(14);
	stack.push(15);
	stack.push(16);
    stack.push(17);
    stack.push(18);
	stack.push(19);
	stack.push(20);
    stack.push(21);
    cout<<"\n";

	stack.pop();
	stack.pop();
	stack.pop();
    cout<<"\n";
    stack.peak();
	cout<<"\n";
	stack.display();
	cout<<"\n";
	return 0;
}

File: Stack implementation using an array.py
maximum=10
arr=[None]*maximum
top=-1
def push(m):
        global maximum,top,arr
        if(top >= maximum-1):
                print("Overflow(insertion not possible): ",m)
        else:
                top+=1
                arr[top]=m
                print("Inserted element in stack:: ",m)
                
def pop():
        global top,arr
        if(top < 0):
                print("Underflow(deletion not possible)")
        else:
                m = arr[top];
                top-=1
                print("Popped element from stack: ",m)
                
def peek():
        if (top < 0):
                print("Stack is Empty")
        else:
                m = arr[top]
                print("Top Element in the stack : ",m)
def display():
        global arr,top
        if(top < 0):
                print("Stack is Empty")
        else:
                print("Elements in the stack : ")
                while(top>=0):
                        m = arr[top]
                        top-=1
                        print(m)

push(10)
push(11)
push(12)
push(13)
push(14)
push(15)
push(16)
push(17)
push(18)
push(19)
push(20)
push(21)
print(" ")
pop()
pop()
pop()
print(" ")
peek() 
print(" ")
display()
